services/worker-manager/test/fake-aws.js

```diff
@@ -1,46 +1,42 @@
-const assert = require('assert');
-
 module.exports = {
   EC2: {
-    runInstances: ({defaultLaunchConfig, TagSpecifications, UserData}) => launchConfig => {
-      assert.deepStrictEqual(launchConfig,
-        {
-          ...defaultLaunchConfig.launchConfig,
-          MinCount: launchConfig.MinCount, // this is estimator's functionality, no need to test this here
-          MaxCount: launchConfig.MaxCount,
-          TagSpecifications,
-          UserData: Buffer.from(JSON.stringify(UserData)).toString('base64'),
+    runInstances: ({defaultLaunchConfig, TagSpecifications, UserData}) => {
+      const fake = launchConfig => {
+        // note that asserting in this function will not cause a test failure, as the error is logged
```

Comment:
I just tried it and I saw an assertion failure. I think the main problem was that the `instance tags in launch spec - should merge them with our instance tags` test case wasn't implemented rather than the mock design

Reply:
The above means, can we at least remove the comment? It doesn't seem to be true to me, like I said above. (I also don't quite see the need to rewrite the mock tbh)